scripts/exregional_novarcldanl.sh

```diff
@@ -0,0 +1,315 @@
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun"
+  ;;
+#
+"ODIN")
+#
+  module list
+
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun -n ${PE_MEMBER01}"
+  ;;
+#
+esac
+#
+#-----------------------------------------------------------------------
+#
+# Extract from CDATE the starting year, month, day, and hour of the
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+set -x
+START_DATE=`echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/'`
+YYYYMMDDHH=`date +%Y%m%d%H -d "${START_DATE}"`
+JJJ=`date +%j -d "${START_DATE}"`
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+#
+#-----------------------------------------------------------------------
+#
+# Get into working directory and define fix directory
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for radar tten process ..."
+
+workdir=${WORKDIR}
+cd_vrfy ${workdir}
+
+fixdir=$FIXgsi
+print_info_msg "$VERBOSE" "fixdir is $fixdir"
+pwd
+
+#
+#-----------------------------------------------------------------------
+#
+# link or copy background and grid configuration files
+#
+#-----------------------------------------------------------------------
+
+cp_vrfy ${fixdir}/fv3_akbk                               fv3_akbk
+cp_vrfy ${fixdir}/fv3_grid_spec                          fv3_grid_spec
+
+bkpath=${CYCLE_DIR}/INPUT
+if [ -w ${bkpath}/gfs_data.tile7.halo0.nc ]; then  # Use background from INPUT
+  ln_vrfy -s ${bkpath}/sfc_data.tile7.halo0.nc      fv3_sfcdata
+  ln_vrfy -s ${bkpath}/gfs_data.tile7.halo0.nc      fv3_dynvars
+  ln_vrfy -s ${bkpath}/gfs_data.tile7.halo0.nc      fv3_tracer
+else                                               # Use background from RESTART
+  ln_vrfy -s ${bkpath}/fv_core.res.tile1.nc         fv3_dynvars
+  ln_vrfy -s ${bkpath}/fv_tracer.res.tile1.nc       fv3_tracer
+  ln_vrfy -s ${bkpath}/sfc_data.nc                  fv3_sfcdata
+fi
+
+#
+#-----------------------------------------------------------------------
+#
+# link/copy observation files to working directory
+#
+#-----------------------------------------------------------------------
+
+PROCESS_BUFR_PATH=${CYCLE_DIR}/process_bufr
+
+obs_file=${PROCESS_BUFR_PATH}/LightningInFV3LAM.dat
+if [ -r "${obs_file}" ]; then
+   cp_vrfy "${obs_file}" "LightningInFV3LAM.dat"
+else
+   print_info_msg "$VERBOSE" "Warning: ${obs_file} does not exist!"
+fi
+
+obs_file=${PROCESS_BUFR_PATH}/NASALaRC_cloud4fv3.bin
+if [ -r "${obs_file}" ]; then
+   cp_vrfy "${obs_file}" "NASALaRC_cloud4fv3.bin"
+else
+   print_info_msg "$VERBOSE" "Warning: ${obs_file} does not exist!"
+fi
+
+obs_file=${PROCESS_BUFR_PATH}/fv3_metarcloud.bin
+if [ -r "${obs_file}" ]; then
+   cp_vrfy "${obs_file}" "fv3_metarcloud.bin"
+else
+   print_info_msg "$VERBOSE" "Warning: ${obs_file} does not exist!"
+fi
+
```

Comment:
Please consider making the above section a loop to reduce the number of repeats. It's easier to maintain in the long run.

```

for obs_file in ${obs_files[@]} ; do
  if [ -r "${obs_file}" ]; then
     cp_vrfy "${obs_file}" $WORKDIR
  else
     print_info_msg "$VERBOSE" "Warning: ${obs_file} does not exist!"
  fi
done
```

Reply:
Need more change here to make it work. Most of the observation files used in DA code have different name from their generation process. Need to pair the observation file name with the the used in the DA first.